File: front-end/src/types/QuestionResponseDTO.ts
import type { QuestionConfigDTO } from "./QuestionConfigDTO";
import type { QuestionOptionResponseDTO } from "./QuestionOptionResponseDTO";
import type { QuestionType } from "./QuestionType";

export interface QuestionResponseDTO {
  id: number;
  type: QuestionType;
  questionText: string;
  explanation?: string;
  imageUrl?: string;
  videoUrl?: string;
  points: number;
  order: number;
  isRequired: boolean;
  config?: QuestionConfigDTO;
  options: QuestionOptionResponseDTO[];
}

File: back-end/src/main/java/com/meutcc/backend/content/question/QuestionResponseDTO.java
package com.meutcc.backend.content.question;

import com.meutcc.backend.content.question.dtos.QuestionOptionResponseDTO;

import java.util.List;
public record QuestionResponseDTO(
        Long id,
        String statement,
        QuestionType type,
        Integer points,
        List<QuestionOptionResponseDTO> options,
        String explanation
) {}